layer1/executor/tasks/base_task.go

```diff
@@ -0,0 +1,165 @@
+package tasks
+
+import (
+	"context"
+	"errors"
+
+	"github.com/alicenet/alicenet/consensus/db"
+	"github.com/alicenet/alicenet/layer1"
+	"github.com/alicenet/alicenet/layer1/transaction"
+	"github.com/sirupsen/logrus"
+)
+
+// TaskAction is an enumeration indicating the actions that the scheduler
+// can do with a task during a request:
+type TaskAction int
+
+// The possible actions that the scheduler can do with a task during a request:
+// * Kill          - To kill/prune a task type immediately
+// * Schedule      - To schedule a new task
+const (
+	Kill TaskAction = iota
+	Schedule
+)
+
+func (action TaskAction) String() string {
+	return [...]string{
+		"Kill",
+		"Schedule",
+	}[action]
+}
+
+type TaskResponse struct {
+	Id  string
+	Err error
+}
+
+type TaskRequest struct {
+	Action TaskAction
+	Task   Task
+}
+
+func NewScheduleTaskRequest(task Task) TaskRequest {
+	return TaskRequest{Action: Schedule, Task: task}
+}
+
+func NewKillTaskRequest(task Task) TaskRequest {
+	return TaskRequest{Action: Kill, Task: task}
+}
+
+type BaseTask struct {
+	Name                string                        `json:"name"`
+	AllowMultiExecution bool                          `json:"allowMultiExecution"`
```

Comment:
You should use this flag in case you don't want to allow more than one execution of a task type per time. E.g if there's a RegistrationTask scheduled, and the flag is false, no other RegistrationTask can be scheduled while the first task finishes or it's pruned/killed.